src/nodes/symlink.rs

```diff
@@ -82,4 +82,10 @@ impl Node for Symlink {
 
         Ok(state.attr)
     }
+
+    fn readlink(&self) -> NodeResult<PathBuf> {
+        let state = self.state.lock().unwrap();
+
+        Ok(fs::read_link(state.underlying_path.as_ref().unwrap())?)
```

Comment:
also here

Reply:
Same as for files. Shouldn't have been optional in the first place.